From 20e72f4d6fa19dcb3491a6bcd1cadd0f1a5d7179 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:54:45 -0400
Subject: [fs] ext4: delete path dealloc code in ext4_ext_handle_uninitialized_extents

Message-id: <1414058125-4183-132-git-send-email-lczerner@redhat.com>
Patchwork-id: 98242
O-Subject: [RHEL 7.1 PATCH v2 131/171] ext4: delete path dealloc code in ext4_ext_handle_uninitialized_extents
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Eric Whitney <enwlinux@gmail.com>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 31cf0f2c3195258f83adabf1a71a782a92b8268a

 Code deallocating the extent path referenced by an argument to
 ext4_ext_handle_uninitialized_extents was made redundant with identical
 code in its one caller, ext4_ext_map_blocks, by commit 3779473246.
 Allocating and deallocating the path in the same function also makes
 the code clearer.

 Signed-off-by: Eric Whitney <enwlinux@gmail.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/extents.c b/fs/ext4/extents.c
index 052b27e..6450498 100644
--- a/fs/ext4/extents.c
+++ b/fs/ext4/extents.c
@@ -4005,10 +4005,6 @@ out1:
  map->m_pblk = newblock;
  map->m_len = allocated;
 out2:
- if (path) {
-  ext4_ext_drop_refs(path);
-  kfree(path);
- }
  return err ? err : allocated;
 }
 
@@ -4208,7 +4204,7 @@ int ext4_ext_map_blocks(handle_t *handle, struct inode *inode,
     err = ret;
    else
     allocated = ret;
-   goto out3;
+   goto out2;
   }
  }
 
@@ -4489,7 +4485,6 @@ out2:
   kfree(path);
  }
 
-out3:
  trace_ext4_ext_map_blocks_exit(inode, flags, map,
            err ? err : allocated);
  ext4_es_lru_add(inode);
-- 
1.7.1